dhcpv4/bsdp/types.go

```diff
@@ -0,0 +1,62 @@
+package bsdp
+
+import "github.com/insomniacslk/dhcp/dhcpv4"
+
+// Options (occur as sub-options of DHCP option 43).
+const (
+	OptionMessageType dhcpv4.OptionCode = iota + 1
+	OptionVersion
+	OptionServerIdentifier
+	OptionServerPriority
+	OptionReplyPort
+	OptionBootImageListPath // Not used
+	OptionDefaultBootImageID
+	OptionSelectedBootImageID
+	OptionBootImageList
+	OptionNetboot1_0Firmware
+	OptionBootImageAttributesFilterList
+	OptionShadowMountPath dhcpv4.OptionCode = 128
+	OptionShadowFilePath  dhcpv4.OptionCode = 129
+	OptionMachineName     dhcpv4.OptionCode = 130
+)
+
+// Versions
+var (
+	Version1_0 = []byte{1, 0}
+	Version1_1 = []byte{1, 1}
+)
+
+//  message types
+const (
+	MessageTypeList byte = iota + 1
+	MessageTypeSelect
+	MessageTypeFailed
+)
+
+// Different types of BootImages - e.g. for different flavors of macOS.
+const (
+	BootImageTypeMacOS9 byte = iota
+	BootImageTypeMacOSX
+	BootImageTypeMacOSXServer
+	BootImageTypeHardwareDiagnostics
```

Comment:
can you define a type `BootImageType`? I.e.

`type BootImageType byte`

and use this throughout the code? It will require more casting around, but it will also make the code safer against erroneous uses of the `ImageType` field that is simply `byte`

Reply:
I did try that, but it was a lot more annoying due to all the casting. If you think that's better, can do.

Reply:
This would be a lot easier if Go supported enums...

Reply:
no that's fine, it's not worth the effort